### PR TITLE
📄 Architecture requirements

### PR DESCRIPTION
**Pull request by swashcap**
* **Problem:** The repository doesn’t have an architecture requirements document!
* **Solution:** Add it from the internal Google doc. Do some cleanup.
* **Testing:** n/a

### CODE REVIEW

File: design/architecture-requirements.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    1. Debugging tools (Example: coinstac-simulator) should exist to allow authors to easily test their algorithms and simulate their use within a COINSTAC-like system
+    2. COINSTAC should perform automated checks against authors’ code to ensure it works within the system, and alert them to errors or potential problems.
+5. Should have an input and output specifications (see [MRN-Code/coinstac#12](https://github.com/MRN-Code/coinstac/issues/12)) to aid in composability and reproducibility:
+    1. Should be able to specify inputs: the computation dictates what it needs to run. This could be a collection of FreeSurfer files, a range of DICOM files, a CSV full of metadata, etc. This may also include inputs from previous computations in the pipeline (Example: second computation needs the original files for meta-analysis).
+    2. Should be able to specify outputs: the computation declares its results. In some cases this may be raw JSON, or a table, a graph, an image, a file path, or a 3D visualization.
+    3. Should easily differentiate between local and remote i/o.???
```
Comment (swashcap) [inline design/architecture-requirements.md]: Any idea how we can clarify this item?

**Comment by swashcap** [comment]
One question on a pipeline list item. I’d say this is good-to-go otherwise.

File: design/architecture-requirements.md (hunk 1 above)
Comment (rssk) [reply design/architecture-requirements.md]: 'Should have provisions for remote and local specifications.'

I think the original is poorly worded, i/o makes me think of actual data, not a specification for the data. In the actual specification, it'll look more like

```
  "local": {
  },
  "remote:" {
  }
```

Which I think is explanatory enough?

Comment (swashcap) [reply design/architecture-requirements.md]: Yeah, that makes perfect sense. I’ll update.

Comment (swashcap) [reply design/architecture-requirements.md]: Addressed in 292aef6502e3f374b68eebb5d2112e9e887ea2ec.

**Review by rssk — APPROVED** [review]

**Pull request merged**